xdsl/dialects/scf.py

```diff
@@ -689,6 +697,107 @@ def parse(cls, parser: Parser) -> Self:
         return op
 
 
+@irdl_op_definition
+class IndexSwitchOp(IRDLOperation):
+    name = "scf.index_switch"
+
+    arg = operand_def(IndexType)
+    cases = attr_def(DenseArrayBase)
+
+    output = var_result_def(AnyAttr())
+
+    default_region = region_def("single_block")
+    case_regions = var_region_def("single_block")
+
+    def __init__(
+        self,
+        arg: Operation | SSAValue,
+        cases: DenseArrayBase,
+        default_region: Region,
+        case_regions: Sequence[Region],
+        result_types: Sequence[Attribute],
+        attr_dict: dict[str, Attribute] | None = None,
+    ):
+        attributes: dict[str, Attribute] = attr_dict or {}
```

Comment:
```suggestion
        attributes = attr_dict or if attr_dict is not None else {}
```

Reply:
Similar to other pr, I believe python's `or` is lazy and so the original should be ok

Reply:
that's not my point, it's that if it's empty it's still Falsey, so it'll create a new one. Also, I think it's bad style to do Falsey things in Python when None is involved, partially for this reason.